src/main/cpp/core/Atlas.h

```diff
@@ -7,35 +7,58 @@
  */
 
 #include <cstddef>
+#include <iostream>
 #include <map>
+#include <string>
 #include <utility>
 #include "core/ClusterType.h"
 #include "geo/GeoPosition.h"
 
 namespace stride {
-
 /**
- * A map from clusters, identified by their ID and type, to geopositions.
+ * A map storing geopositions for clusters, and town data for geopositions.
  */
 class Atlas
 {
 public:
-	using Key = std::pair<std::size_t, ClusterType>;
-	using Map = std::map<Key, geo::GeoPosition>;
+	struct Town
+	{
+		std::string name;
+		unsigned int size;
+		static std::size_t newId;
```

Comment:
Does this have to be `public`? I figure that making this `private` would reduce the odds of someone secretly changing `newId`'s value and messing up all the IDs.